charts/argo-cd/templates/argocd-configs/cluster-secrets.yaml

```diff
@@ -20,8 +20,11 @@ type: Opaque
 stringData:
   name: {{ required "A valid .Values.configs.clusterCredentials[].name entry is required!" .name }}
   server: {{ required "A valid .Values.configs.clusterCredentials[].server entry is required!" .server }}
-  {{- with .namespaces }}
-  namespaces: {{ . }}
+  {{- if .namespaces }}
+  namespaces: {{ .namespaces }}
+  {{- if .clusterResources }}
+  clusterResources: {{ .clusterResources }}
+  {{- end }}
```

Comment:
It looks like `.clusterResources` will only be used if `.namespaces` is also set. Is this the intended effect.

Reply:
Yes, this is intended since `.clusterResources` is only used if `.namespaces` list is not empty.

- https://github.com/argoproj/argo-cd/pull/6581
- https://github.com/argoproj/argo-cd/blob/v2.2.5/pkg/apis/application/v1alpha1/openapi_generated.go#L1445